backend/src/types.ts

```diff
@@ -228,18 +234,78 @@ export type NotebookError = {
 export type NotebookStatus = "Importing" | "Validating" | "Succeeded" | "Failed";
 
 export type Notebook = {
-  name: string;
-  url: string;
-  description?: string;
+  id: string;
   phase?: NotebookStatus;
   user?: string;
   uploaded?: Date;
+  error?: NotebookError;
+} & NotebookCreateRequest & NotebookUpdateRequest;
+
+export type NotebookCreateRequest = {
+  name: string;
+  url: string;
+  description?: string;
+}
+
+export type NotebookUpdateRequest = {
+  id: string;
   visible?: boolean;
   packages?: NotebookPackage[];
```

Comment:
Can this be deleted if we have software?

Reply:
I don't think so. There are 2 distinct content annotatations available in JH per ImageStream, see: 

https://github.com/opendatahub-io/odh-manifests/blob/9ddfec0e5950203acb17d8560378b3c9d67de57e/jupyterhub/notebook-images/overlays/additional/elyra-notebook-imagestream.yaml#L16-L17

`opendatahub.io/notebook-software` maps to the `software` field here and `opendatahub.io/notebook-python-dependencies` maps to `packages. See also https://odh-io.slack.com/archives/C037Q2HLDED/p1648114485074139?thread_ts=1647632733.034829&cid=C037Q2HLDED